vaadin-spreadsheet-flow-parent/vaadin-spreadsheet-flow/src/main/java/com/vaadin/flow/component/spreadsheet/Spreadsheet.java

```diff
@@ -3316,7 +3316,7 @@ public void addMergedRegion(CellRangeAddress region)
 
     private void createMergedRegionIntoSheet(CellRangeAddress region) {
         Sheet sheet = getActiveSheet();
-        int addMergedRegionIndex = sheet.addMergedRegion(region);
```

Comment:
Seems that the index received from the POI API (`addMergedRegion`) does not work well as an index for the local `mergedRegions` list:
- The index from `addMergedRegion` always increases when new regions are added, even if some regions have been removed earlier.
- The local `mergedRegion` always reflects the true set of merged regions